network/report.yml

```diff
@@ -0,0 +1,68 @@
+---
+# Re-write of: 
+# https://github.com/network-automation/toolkit/blob/master/playbooks/network_report.yml
+# https://github.com/network-automation/toolkit/blob/master/roles/build_report/tasks/main.yml
+
+- name: Collect facts
+  hosts: routers
+  gather_facts: false
+
+  tasks:
+    - name: Gather all network resource and minimal legacy facts [Cisco IOS]
+      cisco.ios.ios_facts:
+        gather_subset: min
+        gather_network_resources: all
+      when: ansible_network_os == 'cisco.ios.ios'
+
+    - name: Gather all network resource and minimal legacy facts [Cisco NX-OS]
+      cisco.nxos.nxos_facts:
+        gather_subset: min
+        gather_network_resources: all
+      when: ansible_network_os == 'cisco.nxos.nxos'
+
+    - name: Gather all network resource and minimal legacy facts [Cisco IOS XR]
+      cisco.iosxr.iosxr_facts:
+        gather_subset: min
+        gather_network_resources: all
+      when: ansible_network_os == 'cisco.iosxr.iosxr'
+
+    # The dig lookup requires the python 'dnspython' library  
+    # - name: Resolve IP address
+    #   ansible.builtin.set_fact:
+    #     ansible_host: "{{ lookup('community.general.dig', inventory_hostname)}}"
+
+- hosts: "{{ HOSTS }}"
+  become: yes
+  vars:
+    report_server: node1
+    #web_path: /var/www/html/reports/
+    web_path: /var/www/html
+  
+  tasks:
+    - ansible.builtin.include_role:
+        name: demo.patching.build_report_network
+
+    - block:
```

Comment:
In the block you can reuse the report_server role to ensure apache is installed correctly followed by creating the network report then updating the landing page. Something like this?

```
- include_role:
        name: "{{ item }}"
      loop:
        - demo.patching.report_server
        - demo.patching.build_report_network
        
    - include_role:
        name: demo.patching.report_server
        tasks_from: linux_landing_page
```